src/EditorFeatures/CSharpTest/EditAndContinue/StatementEditingTests.cs

```diff
@@ -9170,6 +9170,178 @@ public void TupleInDelegate()
                 "Update [(int, int) x]@35 -> [(int, int) y]@35");
         }
 
-        #endregion 
+        #endregion
+
+        #region Nullables
+
+        [Fact]
+        public void AddNullCheckBeforeAccessingNullable()
+        {
+            string src1 = @"
+using System;
+
+class C
+{
+    static T id<T>(T t) => t;
+    static T F<T>(Func<T> f) => f();
+
+    static void M(string? x)
+    {
+        var y1 = new { A = id(x) };
+        var y2 = F(() => new { B = id(x) });
+        var z = new Func<string>(() => y1.A + y2.B);
+    }
+}";
+            string src2 = @"
+using System;
+
+class C
+{
+    static T id<T>(T t) => t;
+    static T F<T>(Func<T> f) => f();
+
+    static void M(string? x)
+    {
+        if (x is null) throw new Exception();
+        var y1 = new { A = id(x) };
+        var y2 = F(() => new { B = id(x) });
+        var z = new Func<string>(() => y1.A + y2.B);
+    }
+}";
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                // (9,25): error CS8652: The feature 'nullable reference types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     static void M(string? x)
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "?").WithArguments("nullable reference types").WithLocation(9, 25));
```

Comment:
What's the point of checking the error? You should set the test up so that the error is not reported.

Reply:
yes, total mess. Sorry!

---
In reply to: [301277114](https://github.com/dotnet/roslyn/pull/36770#discussion_r301277114) [](ancestors = 301277114)